internal/collector/instance.go

```diff
@@ -144,22 +147,37 @@ func AddToPod(
 
 // startCommand generates the command script used by the collector container
 func startCommand(includeLogrotate bool) []string {
-	var startScript = `
-/otelcol-contrib --config /etc/otel-collector/config.yaml
-`
-
+	var logrotateCommand string
 	if includeLogrotate {
-		startScript = `
-/otelcol-contrib --config /etc/otel-collector/config.yaml &
+		logrotateCommand = `logrotate -s /tmp/logrotate.status /etc/logrotate.d/logrotate.conf`
+	}
+
+	var startScript = fmt.Sprintf(`
+OTEL_PIDFILE=/tmp/otel.pid
+
+start_otel_collector() {
+	echo "Starting OTel Collector"
+	/otelcol-contrib --config %s/config.yaml &
+	echo $! > $OTEL_PIDFILE
+}
+start_otel_collector
 
 exec {fd}<> <(:||:)
 while read -r -t 5 -u "${fd}" ||:; do
-	logrotate -s /tmp/logrotate.status /etc/logrotate.d/logrotate.conf
+	%s
```

Comment:
🌱 Perhaps this could always be `[[ -r /etc/logrotate.d/logrotate.conf ]] && logrotate ...` and the choice to rotate or not would be at the mounting of the config.